### PR TITLE
waypoint login -from-kubernetes

### PR DESCRIPTION
**Pull request by mitchellh**
This introduces `waypoint login -from-kubernetes` that performs automatic authentication after a Helm-based install. This looks up the Waypoint service, gets the token from the Kubernetes secret, and completes the login.

### CODE REVIEW

File: internal/cli/login.go [hunk 1]
```diff
@@ -319,6 +502,26 @@ waypoint login https://example.com
...
 double-check the token and try again.
+`
+
+	errK8STokenEmpty = `
+The Waypoint token in the Kubernetes secret is empty. This is usually
+for one of two reasons. First, the Waypoint server may not be bootstrapped.
+After installing Waypoint on Kubernetes, it takes a few minutes for Waypoint
+to bootstrap itself.
+
+If Waypoint is already bootstrapped, its possible the server administrator
```
Comment (briancain) [inline internal/cli/login.go:513]:
```suggestion
If Waypoint is already bootstrapped, it's possible the server administrator
```

File: internal/cli/login.go [hunk 2]
```diff
@@ -239,6 +316,81 @@ func (c *LoginCommand) loginOIDC() (string, int) {
 	return respToken.Token, 0
 }
 
+func (c *LoginCommand) loginK8S(ctx context.Context) (string, int) {
```
Comment (briancain) [inline internal/cli/login.go:319]: Adding this makes me think we should have some kind of Interface for our authentication functions here for a more rigid way to define these auth funcs, thoughts? I could see us adding more in the future.

**Review by briancain — APPROVED** [review]
🔒 🎉  Make sense! 👍🏻 Had a minor comment that we could maybe address here, or maybe leave as a TODO here for the future if we add more authentication functions to our login.

File: internal/cli/login.go (hunk 2 above)
Comment (mitchellh) [reply internal/cli/login.go:319]: Its already typed to `func(context.Context) (string,int)` and in the switch above, so it sort of already is? I don't think we need a separate full Go interface cause they can't stand on their own (they require the login command metadata).

Comment (briancain) [reply internal/cli/login.go:319]: Cool cool, makes sense!

**Pull request merged**